lasagne/layers/conv.py

```diff
@@ -117,23 +109,22 @@ class Conv1DLayer(Layer):
         An integer or a 1-element tuple specifying the stride of the
         convolution operation.
 
-    border_mode : str, one of 'valid', 'full', 'same'
-        A string indicating the convolution border mode.
+    pad : {0, (0,), 'full', 'same'} (default: 0)
```

Comment:
I think using `theano_extensions` would be the best option, so that it can have the same semantics as the other ConvLayers. Then when Theano catches up, it should be a simple speedup with no change visible to the user. It would be nice if you could reuse PadLayer, but I think you're right that returning two layers would be confusing.